Make the protected object public to use its functions elsewhere.

alarshi opened the pull request:
We use the object `slab_boundary` in our material model to access `slab_depths` and `slab_thickness`. In order to do this, this PR allows public access for the object. 

### Before your first pull request:

* [x] I have read the guidelines in our [CONTRIBUTING.md](../blob/main/CONTRIBUTING.md) document.

### For all pull requests:

* [x] I have followed the [instructions for indenting my code](../blob/main/CONTRIBUTING.md#making-aspect-better).

bangerth commented:
If you make it `public`, you also allow other places in the code to *change* the variable.

My preference would be to keep it protected and instead add an accessor function
```
  const Utilities::AsciiDataBoundary<dim> &get_slab_boundary() const;
```
This way, other places can access the data, but because they only get a `const` reference, they can't change it.

Separate question: Why was it `protected` to begin with, rather than `private`?

alarshi commented:
Thank you for the review! :)  I addressed your comments in the latest commit. 
I think `Protected` access instead of `Private` was a carryover mistake. I have corrected it now.

bangerth reviewed (approved):
Yes, looks good. Thank you!
